cluster/interface.go

```diff
@@ -0,0 +1,19 @@
+package cluster
+
+const (
+	// EtcdLeaderElector is the id used when constructing an
+	// etcdLeaderElector instance through the factory.
+	EtcdLeaderElector = "etcdLeaderElector"
+)
+
+// LeaderElector is a general interface implementing basic leader elections
+// in a clustered environment.
+type LeaderElector interface {
```

Comment:
Should this also be extended to add the `Leader` method that's present on the current default implementation?

Reply:
Done.